combat-trainer.lic

```diff
@@ -3837,9 +3837,11 @@ class GameState
 end
 
 before_dying do
+  @tk_ammo = get_settings.tk_ammo
```

Comment:
❌ You can't use `get_settings` like this because it won't load additional YAML files, only the base.

See https://github.com/rpherbig/dr-scripts/blob/master/combat-trainer.lic#L1040 for an example of using the passed in settings.

I'm pretty sure you can't access `@tk_ammo` here since it's not in the `SpellProcess` class. I'm also pretty sure we can't pass anything into `before_dying` since, if I remember right, that's a Lich thing.

My suggestion is to make `tk_ammo` a global instead of instance var in `SpellProcess`. If you have another idea, I'm all ears. Slack may also have some suggestions.

Reply:
I am almost certain i tested the @tk_ammo = settings.tk_ammo and it didn't work. I will look at making it global. and test that.

Reply:
**settings.tk_ammo** will not work because settings is a class variable and **before_dying** is out of scope.